courseware/api.py

```diff
@@ -564,12 +582,24 @@ def retry_failed_edx_enrollments():
         course_run = enrollment.run
         try:
             enroll_in_edx_course_runs(user, [course_run])
+        except EdxApiEnrollErrorException as exc:
+            # Check if user is already actively enrolled
+            edx_enrollment = get_enrollment(user, course_run)
+            if edx_enrollment and edx_enrollment.is_active:
```

Comment:
Less likely, but would it be a good idea to check for enrollment mode here as well?

 I became curious when I saw our [enroll_in_edx_course_runs](https://github.com/mitodl/mitxpro/blob/master/courseware/api.py#L499) method which initially tries to enroll the user in `id-professional` mode and upon failure, with this mode, it then tries to enroll the user in `Audit` mode. So this made me wonder if we should check the mode before marking the user as `edx_enrolled`. 

I also think `edx_enrolled` is independent of the enrollment mode anyway in which case I think this logic is fine.

Reply:
Thanks, I added mode as an additional check